### PR TITLE
Add api historical balances

### PR DESCRIPTION
**Pull request by tspenov**
#### Summary
`for address`:

```graphql
 {
    historicalBalance(
        address: "0xe76fe52a251c8f3a5dcd657e47a6c8d16fdf4bfa",
        from: "2018-11-01T00:00:00Z",
        to: "2018-11-15T00:00:00Z",
        interval: "1d"
    ){
        datetime,
        balance
    }
}
```

`or for erc20 project:`

```graphql
{
    historicalBalance(
      	slug: "santiment",
        address: "0x5e575279bf9f4acf0a130c186861454247394c06",
        from: "2018-07-01T00:00:00Z",
        to: "2018-11-15T00:00:00Z",
        interval: "1d"
    ){
        datetime,
	balance
    }
}
```

```javascript
{
  "data": {
    "historicalBalance": [
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-01T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-02T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-03T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-04T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-05T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-06T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-07T00:00:00Z"
      },
      {
        "balance": 803.1942152452249,
        "datetime": "2018-11-08T00:00:00Z"
      },
      {
        "balance": 792.1938162452249,
        "datetime": "2018-11-09T00:00:00Z"
      },
      {
        "balance": 792.1938162452249,
        "datetime": "2018-11-10T00:00:00Z"
      },
      {
        "balance": 792.1938162452249,
        "datetime": "2018-11-11T00:00:00Z"
      },
      {
        "balance": 789.3236482452248,
        "datetime": "2018-11-12T00:00:00Z"
      },
      {
        "balance": 789.3236482452248,
        "datetime": "2018-11-13T00:00:00Z"
      },
      {
        "balance": 789.3236482452248,
        "datetime": "2018-11-14T00:00:00Z"
      },
      {
        "balance": 789.3236482452248,
        "datetime": "2018-11-15T00:00:00Z"
      }
    ]
  }
}
```

### CODE REVIEW

File: lib/sanbase/clickhouse/eth_transfers.ex [hunk 1]
```diff
@@ -296,6 +309,90 @@ defmodule Sanbase.Clickhouse.EthTransfers do
...
+
+    dt_round = datetime_rounding_for_interval(interval)
+
+    query = """
+    SELECT dt, runningAccumulate(state) AS total_balance FROM (
+      SELECT dt, sumState(value) AS state FROM (
+        SELECT
+          toUnixTimestamp(#{dt_round}) as dt, "from" AS address, sum(-value / #{@eth_decimals}) AS value
+        FROM #{@table}
+        PREWHERE "from" = ?1
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/eth_transfers.ex]: are the quotes needed here?

File: lib/sanbase_web/graphql/schema.ex [hunk 2]
```diff
@@ -602,6 +604,16 @@ defmodule SanbaseWeb.Graphql.Schema do
...
     end
+
+    @desc "Historical balance for an address"
+    field :historical_balance, list_of(:historical_balance) do
+      arg(:from, non_null(:datetime))
+      arg(:to, non_null(:datetime))
+      arg(:address, non_null(:string))
+      arg(:interval, non_null(:string), default_value: "1d")
+
+      resolve(&ClickhouseResolver.historical_balance/3)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema.ex]: `cache_resolve`

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  alias Sanbase.Model.Project
+
+  @one_hour_seconds 3600
+
+  def historical_balance(
+        _root,
+        args,
+        _resolution
+      ) do
+    calc_historical_balances(args)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Do we need to abstract this?

File: lib/sanbase/clickhouse/common.ex [hunk 4]
```diff
@@ -0,0 +1,56 @@
+defmodule Sanbase.Clickhouse.Common do
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/common.ex]: The functionality here is suitable for the ProjectBalance resolver module

Comment (IvanIvanoff) [reply lib/sanbase/clickhouse/common.ex]: @tspenov it calculates only the historical balance for a given slug (project) so it is basically project historical balance

Comment (tspenov) [reply lib/sanbase/clickhouse/common.ex]: it also calculates without project - only by address. Are you saying the queries in the description should be changed somehow and go through `all_projects` ?

**Pull request merged**